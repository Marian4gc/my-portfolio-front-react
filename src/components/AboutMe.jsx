import React from 'react'
import foto from '../assests/fotoMarian.jpg'
import '../styles/aboutme.css'

function AboutMe() {
    return (
    <div className="container col-xxl-8 px-4 py-5" id="about">
            <div className="row flex-lg-row-reverse align-items-center g-5 py-5">
                <div className="col-10 col-sm-8 col-lg-6">
                    <img src={foto} className="d-block mx-lg-auto img-fluid foto" alt="Bootstrap Themes" width="700" height="500" loading="lazy"/>
                </div>
                <div className="col-lg-6">
                    <h1 className="display-5 fw-bold lh-1 mb-3 who">Who is Marian?</h1>
                        <p className="lead">Hi! I am a woman who has just entered the world of coding and discovered my passion for it. I enjoy identifying and solving problems while continuously learning. Overcoming challenges and bringing my design ideas to life in the form of web applications is what excites me. I have completed an 850-hour Full-Stack Developer Bootcamp, but my curiosity to keep learning and advancing remains insatiable. Feel free to get in touch if you'd like to connect.</p>
                </div>
            </div>
    </div>
    );
}

export default AboutMe
