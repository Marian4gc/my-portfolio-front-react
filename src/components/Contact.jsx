import React from 'react';
// import { Form, } from 'react-bootstrap';
// import { Link } from 'react-router-dom';

import '../styles/contact.css'

function Contact() {



  return (

    <div>
      <div className="container px-4 py-5" id="custom-cards">
        <h2 className="display-5 fw-bold work" id='contact' >Contact</h2>
        <hr className="featurette-divider"></hr>

        <p className="lead mb-4 text-center">Do not hesitate to contact me if you want to know more</p>
      </div>

      <div className="container px-4 py-5 text-center mobil">
        <button data-text="Awesome" className="button">
          <span className="actual-text">&nbsp;mariangarcia1724@gmail.com&nbsp;</span>
          <span className="hover-text" aria-hidden="true">&nbsp;mariangarcia1724@gmail.com&nbsp;</span>
        </button>
      </div>




      {/* <div className="p-3 rounded border border-secondary bg-transparent mx-auto contact">
        <Form>
          <Form.Group controlId="formName">
            <Form.Label>Name</Form.Label>
            <Form.Control type="text" placeholder="Enter your name" required />
          </Form.Group>

          <Form.Group controlId="formEmail">
            <Form.Label>Email address</Form.Label>
            <Form.Control type="email" placeholder="Enter email" required />
          </Form.Group>

          <Form.Group controlId="formMessage">
            <Form.Label>Message</Form.Label>
            <Form.Control as="textarea" rows={3} placeholder="Enter your message" required />
          </Form.Group>

          <div className="d-grid gap-2 d-sm-flex" style={{ textAlign: "center", justifyContent: "center" }}>
            <Link to="#">
              <button type="button" className="btn btn-primary btn-lg px-4 gap-3 submt">Submit</button>
            </Link>
          </div>
        </Form>
      </div> */}
    </div>
  );
}

export default Contact;

