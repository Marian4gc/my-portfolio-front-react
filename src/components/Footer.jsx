import React from 'react';
import { FaLinkedin, FaGithub } from 'react-icons/fa';
import '../styles/footer.css'

function Footer() {
    return (
        <footer>
        <div style={{ display: 'flex', justifyContent: 'center' }}>
            <a href='https://www.linkedin.com/in/mariangarciacabrera/' style={{ margin: '10px', color: 'white' }} >
            <FaLinkedin size='2em' />
            </a>
            <a href='https://github.com/Marian4gc' style={{ margin: '10px', color: 'white' }}>
            <FaGithub size='2em' />
            </a>
        </div>
        <p style={{ textAlign: 'center', margin: '10px' }}>Â© 2023 My Portfolio. All rights reserved.</p>
        </footer>
    );
}

export default Footer;

// import React from 'react';
// import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
// import { faGithub, faLinkedin } from '@fortawesome/free-brands-svg-icons';

// function Footer() {
//     return (
//         <footer>
//         <div className="container">
//             <div className="row">
//             <div className="col-sm-12 text-center">
//                 <ul className="list-inline">
//                 <li className="list-inline-item">
//                     <a href="https://www.linkedin.com" target="_blank" rel="noreferrer"><FontAwesomeIcon icon={faLinkedin} size="2x" /></a>
//                 </li>
//                 <li className="list-inline-item">
//                     <a href="https://github.com" target="_blank" rel="noreferrer"><FontAwesomeIcon icon={faGithub} size="2x" /></a>
//                 </li>
//                 </ul>
//                 <p>&copy; Your Name</p>
//             </div>
//             </div>
//         </div>
//         </footer>
//     );
// }

// export default Footer;


