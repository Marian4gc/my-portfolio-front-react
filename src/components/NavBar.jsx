import React, { useState } from "react";
import { Link } from "react-scroll";
import "../styles/navbar.css";
import cv from "../assests/CV-MARIAN GARCIA-FULLSTACK.pdf";

export default function NavBar() {
  const [isOpen, setIsOpen] = useState(false);

  const [color, setColor] = useState(false);
  const changeColor = () => {
    if (window.scrollY >= 60) {
      setColor(true);
    } else {
      setColor(false);
    }
  };

  window.addEventListener("scroll", changeColor);

  return (
    <>
      <div className={color ? "header header__bg" : "header"}>
        <div className="nav__container">
          <Link to="home" className="nav__logo">
            Marian Garc√≠a
          </Link>
          <div className={`nav__items ${isOpen && "open"}`}>
            <Link
              to="about"
              // spy={true}
              smooth={true}
              offset={-100}
              duration={1000}
              className="nav__links"
            >
              About
            </Link>
            <Link
              to="repos"
              // spy={true}
              smooth={true}
              offset={-100}
              duration={1000}
              className="nav__links"
            >
              Work
            </Link>
            <Link
              to="skills"
              // spy={true}
              smooth={true}
              offset={-100}
              duration={1000}
              className="nav__links"
            >
              Skills
            </Link>
            <Link
              to="contact"
              // spy={true}
              smooth={true}
              offset={-100}
              duration={1000}
              className="nav__links"
            >
              Contact
            </Link>
            <a
              href={cv}
              target="_blank"
              rel="noopener noreferrer"
              className="nav__links"
            >
              CV
            </a>
          </div>
          <div
            className={`nav__toggle ${isOpen && "open"}`}
            onClick={() => setIsOpen(!isOpen)}
          >
            <div className="bar"></div>
          </div>
        </div>
      </div>
    </>
  );
}
