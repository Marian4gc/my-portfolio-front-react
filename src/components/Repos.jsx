import React, { useEffect, useState } from "react";
import 'bootstrap/dist/css/bootstrap.min.css';
import '../styles/repos.css'

function Repos() {
    
const [repo, setRepo] = useState([]);

    useEffect(() => {
        fetch('http://127.0.0.1:8000/apiportfolio/list')
        .then(response => response.json())
        .then(repo => setRepo(repo))
        .catch(error => console.error(error));        
    }, []);
    return (
        <div className="container px-4 py-5" id="custom-cards">
            <h2 className="display-5 fw-bold work">Work</h2>
            <hr className="featurette-divider"></hr>

            <div className="row row-cols-1 row-cols-lg-3 align-items-stretch g-4 py-5" id="repos">
            {repo.map(int => (
                <div className="col" key={int.id}>
                <a href={int.link} target="_blank" rel="noopener noreferrer" style={{ textDecoration: 'none' }}>
                    <div className="card card-cover h-100 overflow-hidden text-bg-dark rounded-4 shadow-lg card-hover" style={{backgroundImage: `url(${int.image})`, backgroundSize: '100% 100%', backgroundRepeat: 'no-repeat', backgroundAttachment: 'contain'}}>
                    <div style={{ width: '100%', height: '100%', position: 'absolute', top: 0, left: 0, zIndex: 1 }} />
                    <div className="d-flex flex-column h-100 p-5 pb-3 text-white text-shadow-1" style={{ position: 'relative', zIndex: 2 }}>
                        <h3 className="mt-5 mb-4 display-6 lh-1 fw-bold name">{int.name}</h3>
                        <ul className="d-flex list-unstyled mt-auto">
                        <li className="d-flex align-items-center me-3">
                            <small className="descp">{int.description}</small>
                        </li>
                        </ul>
                    </div>
                    </div>
                </a>
                </div>    
                ))}
            </div>
        </div>

        );
}

export default Repos;  