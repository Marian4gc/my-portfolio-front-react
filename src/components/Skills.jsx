import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHtml5, faCss3Alt, faJsSquare, faReact, faBootstrap, faSymfony, faFigma, faGitAlt, faGithub, faPhp, faNode, faNpm} from '@fortawesome/free-brands-svg-icons';
import { Col, Row } from 'react-bootstrap';
import "../styles/skills.css";
import { faDatabase } from '@fortawesome/free-solid-svg-icons';
import { faCode } from '@fortawesome/free-solid-svg-icons';
import cypressIcon from '../assests/navbar-brand.svg';
import postmanIcon from '../assests/postman-icon.svg';


const Skills = () => {
    return (
        <div id='skills' className='container-fluid'>
        <div className='d-flex justify-content-center my-5 fw-bold'>
            <h1 className="display-5 fw-bold skills">Skills</h1>
        </div>
            <hr className="featurette-divider"></hr>
        <div className="skills-icons">
            <Row className='text-center'>
                <Col xs={6} md={3} className='p-4'>
                <FontAwesomeIcon icon={faHtml5} size='3x' />
                <h3 className='mt-4 mb-3'>HTML5</h3>
                </Col>
                <Col xs={6} md={3} className='p-4'>
                <FontAwesomeIcon icon={faCss3Alt} size='3x' />
                <h3 className='mt-4 mb-3'>CSS3</h3>
                </Col>
                <Col xs={6} md={3} className='p-4'>
                <FontAwesomeIcon icon={faJsSquare} size='3x' />
                <h3 className='mt-4 mb-3'>JavaScript</h3>
                </Col>
                <Col xs={6} md={3} className='p-4'>
                <FontAwesomeIcon icon={faReact} size='3x' />
                <h3 className='mt-4 mb-3'>React</h3>
                </Col>
                <Col xs={6} md={3} className='p-4'>
                <FontAwesomeIcon icon={faBootstrap} size='3x' />
                <h3 className='mt-4 mb-3'>Bootstrap</h3>
                </Col>
                <Col xs={6} md={3} className='p-4'>
                    <FontAwesomeIcon icon={faSymfony} size='3x' />
                    <h3 className='mt-4 mb-3'>Symfony</h3>
                </Col>
                <Col xs={6} md={3} className='p-4'>
                <FontAwesomeIcon icon={faDatabase} size='3x' />
                <h3 className='mt-4 mb-3'>SQL</h3>
                </Col>
                <Col xs={6} md={3} className='p-4'>
                <FontAwesomeIcon icon={faCode} size='3x' />
                <h3 className='mt-4 mb-3'>Visual Studio Code</h3>
                </Col>
                <Col xs={6} md={3} className='p-4'>
                <FontAwesomeIcon icon={faFigma} size='3x' />
                <h3 className='mt-4 mb-3'>Figma</h3>
                </Col>
                <Col xs={6} md={3} className='p-4'>
                <FontAwesomeIcon icon={faGitAlt} size='3x' />
                <h3 className='mt-4 mb-3'>Git</h3>
                </Col>
                <Col xs={6} md={3} className='p-4'>
                <FontAwesomeIcon icon={faGithub} size='3x' />
                <h3 className='mt-4 mb-3'>GitHub</h3>
                </Col>
                <Col xs={6} md={3} className='p-4'>
                <FontAwesomeIcon icon={faPhp} size='3x' />
                <h3 className='mt-4 mb-3'>PHP</h3>
                </Col>
                <Col xs={6} md={3} className='p-4'>
                <FontAwesomeIcon icon={faNode} size='3x' />
                <h3 className='mt-4 mb-3'>Node</h3>
                </Col>
                <Col xs={6} md={3} className='p-4'>
                <FontAwesomeIcon icon={faNpm} size='3x' />
                <h3 className='mt-4 mb-3'>Npm</h3>
                </Col>
                <Col xs={6} md={3} className='p-4'>
                <img src={cypressIcon} alt="Cypress Icon" className="cypress-icon" />
                <h3 className='mt-4 mb-3'>Cypress</h3>
                </Col>
                <Col xs={6} md={3} className='p-4'>
                <img src={postmanIcon} alt="Cypress Icon" size='3x' className="postman-icon" />
                <h3 className='mt-4 mb-3'>Postman</h3>
                </Col>

            </Row>
        </div> 
        </div>
    );
};

export default Skills;

