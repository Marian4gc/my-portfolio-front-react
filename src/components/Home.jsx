import React from "react";
import { Link } from 'react-scroll';
import "../styles/home.css";
import NavBar from './NavBar';

function Home() {
    return (
        <>
        <NavBar />
        <div className="text_home" id="home">
        <div className="px-4 py-5 my-5 text-center">
            <h2>Hi, my name is</h2>
            <h1 className="display-5 fw-bold">Marian Garc√≠a.</h1>
                    <div className="col-lg-6 mx-auto">
                        <p className="lead mb-4">I love being able to translate my ideas with the code</p>
                        <h3>I am Junior FullStack Developer</h3>
                    </div>
            <div className="d-grid gap-2 d-sm-flex justify-content-sm-center">
                <Link
                    to="repos"
                    smooth={true}
                    offset={-100}
                    duration={1000}
                >
                    <button type="button" className="btn btn-primary btn-lg px-4 gap-3 mira">Check out my work!</button>
                </Link>
            </div>
        </div>
        </div>
        </>
    );
}

export default Home;
