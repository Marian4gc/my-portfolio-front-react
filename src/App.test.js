import { render, screen, waitFor, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import AboutMe from './components/AboutMe';
import Contact from './components/Contact';
import Home from './components/Home';
import Repos from './components/Repos';
import Skills from './components/Skills';
import NavBar from './components/NavBar';
import Footer from './components/Footer';
import renderer from 'react-test-renderer';


test('renders learn react link', () => {
  render(<NavBar />);
  const linkElement = screen.getByText(/Marian García/i);
  expect(linkElement).toBeInTheDocument();
});

test('renders Contact without crashing', () => {
  render(
    <BrowserRouter basename="/">
      <Contact />
    </BrowserRouter>
  );
});

test('renders AboutMe without crashing', () => {
  render(<AboutMe />);
});

test('renders Home without crashing', () => {
  render(<Home />);
});

test('renders Repos without crashing', () => {
  render(<Repos />);
});

test('renders Skills without crashing', () => {
  render(<Skills />);
});

test('renders NavBar without crashing', () => {
  render(<NavBar />);
});

test('renders Footer without crashing', () => {
  render(<Footer />);
});

test('renders a button', () => {
  const { getByText } = render(<button>Check out my work!</button>);
  const button = getByText('Check out my work!');
  expect(button).toBeInTheDocument();
});

describe('AboutMe', () => {
  it('renders the image correctly', () => {
    render(<AboutMe />);
    const imageElement = screen.getByAltText('Bootstrap Themes');
    expect(imageElement).toBeInTheDocument();
    expect(imageElement).toHaveAttribute('src', 'fotoMarian.jpg');
  });
});

test('Home component matches snapshot', () => {
  const component = renderer.create(<Repos />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});



import { MemoryRouter } from 'react-router-dom';



//comprobar que se están haciendo bien las renderizaciones y las peticones de la API
describe('Repos', () => {
  it('comprobar que se reciben bien los datos de la API', async () => {
    const mockRepos = [
      {
        id: 1,
        name: 'Tanjiro', 
        description: 'Maquetación de página web con bootstrap', 
        image: 'https://i.postimg.cc/ZRqcLh1v/Captura-de-pantalla-2023-03-18-004207.png', 
        link: 'https://github.com/Marian4gc/proyectoweb',
      },
      {
        id: 2,
        name: 'Aplicación con React',
        description: 'Aplicación con React en entorno node.',
        image: 'https://i.postimg.cc/6qhZZvyr/Captura-de-pantalla-2023-03-17-141107.png',
        link: 'https://github.com/Marian4gc/formulario-js',
      },
      {
        id: 3,
        name: 'Validación de formulario JS',
        description: 'Validar la información de un formulario con JavaScript',
        image: 'https://i.postimg.cc/X7WQmKWg/Captura-de-pantalla-2023-03-17-135723.png',
        link: 'https://github.com/Marian4gc/vivero-react',
      },
      {
        id: 4,
        name: 'Formulario React',
        description: 'Formulario de React validado, utilizando Formik',
        image: 'https://i.postimg.cc/RhKn7sVn/Captura-de-pantalla-2023-03-18-002814.png',
        link: 'https://github.com/Marian4gc/formulario-react',
      },
      {
        id: 5,
        name: 'AnimApp',
        description: 'AnimApp es una aplicación web de comercio electrónico diseñada para ayudar a las personas a adoptar animales.',
        image: 'https://user-images.githubusercontent.com/115170876/221117608-73375859-b8f3-4489-ab3d-04ef7e3dae47.jpg',
        link: 'https://github.com/Marian4gc/hackathon-animApp',
      },
      {
        id: 6,
        name: 'Aplicación Navidad',
        description: 'Web dedicada a la recepción y a la realización de cartas para todos las niñas y niños del mundo.',
        image: 'https://i.postimg.cc/4NqTz6LG/aplicacion-Navidad.png',
        link: 'https://github.com/Marian4gc/proyectonavidad',
      },
    ];
      
    jest.spyOn(global, 'fetch').mockResolvedValue({
    json: jest.fn().mockResolvedValue(mockRepos),
        });

      render(<Repos />);

      await waitFor(() => expect(global.fetch).toHaveBeenCalled());

      const repoElements = await screen.findAllByRole('link');

    expect(repoElements).toHaveLength(6);
    expect(repoElements[0]).toHaveAttribute('href', mockRepos[0].link);
    expect(repoElements[1]).toHaveAttribute('href', mockRepos[1].link);

    });
});






