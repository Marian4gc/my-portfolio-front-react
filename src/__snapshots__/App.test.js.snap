// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component matches snapshot 1`] = `
<div
  className="container px-4 py-5"
  id="custom-cards"
>
  <h2
    className="display-5 fw-bold work"
  >
    Work
  </h2>
  <hr
    className="featurette-divider"
  />
  <div
    className="row row-cols-1 row-cols-lg-3 align-items-stretch g-4 py-5"
    id="repos"
  />
</div>
`;
